src/dotnet/commands/dotnet-run3/Run3Command.cs

```diff
@@ -49,6 +52,27 @@ public int Start()
                 .ExitCode;
         }
 
+        private void EnsureProjectIsBuilt()
+        {
+            List<string> buildArgs = new List<string>();
+
+            buildArgs.Add("/nologo");
+            buildArgs.Add("/verbosity:quiet");
+
+            if (!string.IsNullOrWhiteSpace(Configuration))
+            {
+                buildArgs.Add($"/p:Configuration={Configuration}");
+            }
+
+            var buildResult = new MSBuildForwardingApp(buildArgs).Execute();
```

Comment:
I'm surprised to see this using MSBuild directly rather than through build3. 

I can't think of a downside at the moment, but what if we introduce some additional logic in build3?